contributors/devel/sig-scheduling/scheduler_benchmarking.md

```diff
@@ -0,0 +1,77 @@
+# Scheduler Benchmarking
+
+Kubernetes scheduler has integration and e2e benchmarks. It is recommended to
+run integration benchmarks at the time of submitting PRs that could potentially
+impact performance of the scheduler and give before and after results with your
+PR submission.
+
+## Running integration benchmarks
+
+To run integration benchmarks use the following command from inside a Kubernetes
+directory. 
+
+```sh
+make test-integration WHAT=./test/integration/scheduler_perf KUBE_TEST_VMODULE="''" KUBE_TEST_ARGS="-run=xxx -bench=BenchmarkScheduling"
+```
+
+These benchmarks are located in `./test/integration/scheduler_perf/scheduler_bench_test.go`. 
+The function names start with `BenchmarkScheduling`. At the beginning of each
+function there are a few lines in the form of:
+
+```go
+	tests := []struct{ nodes, existingPods, minPods int }{
+		{nodes: 100, existingPods: 1000, minPods: 100},
+		{nodes: 1000, existingPods: 1000, minPods: 100},
+		{nodes: 5000, existingPods: 1000, minPods: 1000},
+	}
+```
+
+Each line indicates a test. `nodes` specifies the number of nodes in the test
+cluster. `existingPods` specifies how many pods should be created and scheduled
+as the initialization phase of the benchmark. `minPods` specifies how many pods
+must be created and scheduled as the actual part of benchmarking.
+You may add other items to the above array and run the benchmarks again. For 
+example, if you want to measure performance in a 2000 node cluster when scheduling
```

Comment:
I don't feel that changing the test cases should be part of this documentation. Or at the very least, it should not be under the heading "running integration benchmarks"

Reply:
Perhaps what we need is to have a section in this doc for running benchmarks and another section for writing new ones or modifying the test cases. Any literal references to test names, files, or code may get out of sync with this document, so let's think about whether that can be avoided.

Reply:
I am not sure if I agree. I myself usually run benchmarks with a lot larger number of pods in order to minimize other overheads. At the same time, we haven't been able to add all those configurations in the code because they timeout in our CI/CD.
I agree that literal references to test names and files requires additional effort in keeping them in sync with documents, but at this point we don't have a much better way of doing this without losing usefulness of the doc.

Reply:
@misterikkit I see your point. From a viewpoint of usefulness, having concrete examples is really helpful so I'd be in favor of leaving this, perhaps with a disclaimer and a link to current test files?

Reply:
Based on how I see this get used, it means that people who want to test extra large scenarios have to carry their own patches around, and there is no guarantee that those patches match what other contributors are using for test. Maybe these scenarios are unique enough that it's not important.

I *am* in favor of having documentation saying, "here is how to add a test case with lots of nodes/pods".